### ISSUE TITLE
Fish: `brew edit` does not respect $EDITOR

### ISSUE THREAD

**Issue by LEARAX** [Homebrew/brew#5220]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

`brew doctor`:
```
Will ) brew doctor
Your system is ready to brew.
```

`brew config`:
```
Will ) brew config
HOMEBREW_VERSION: 1.8.0-76-gcbb2e1b
ORIGIN: https://github.com/Homebrew/brew
HEAD: cbb2e1b3dd63e1b9621c67b619aad8c914772041
Last commit: 10 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 92bf96e55a50a81baad43434ada0053a67b8a45d
Core tap last commit: 24 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_NO_ANALYTICS_THIS_RUN: 1
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.19.1 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.1, 1.8.0_92
macOS: 10.14.1-x86_64
CLT: 10.0.0.0.1.1535735448
Xcode: 10.0
XQuartz: 2.7.11 => /opt/X11
```

To help us debug your issue please explain:
- What you were trying to do (and why)
I was attempting to edit the build script for cataclysm, in order to modify some settings.
- What happened (include command output)
As seen below, Homebrew opens atom instead of neovim.

```
Will ) brew edit cataclysm
Editing /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cataclysm.rb
Warning: Using atom because no editor was set in the environment.
This may change in the future, so we recommend setting EDITOR,
or HOMEBREW_EDITOR to your preferred text editor.
Will ) echo $EDITOR
/usr/local/bin/nvim
Will ) echo $HOMEBREW_EDITOR

Will )
```
Running `/usr/local/bin/nvim` behaves as expected. Neovim behaves normally otherwise. Switching to bash, as shown below, results in desired behavior.
```
Will ) bash
[Will:Lanyard][~]$ brew edit cataclysm
Editing /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cataclysm.rb
[Will:Lanyard][~]$ echo $EDITOR
/usr/local/bin/nvim
[Will:Lanyard][~]$
```

- What you expected to happen
I expect the file `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cataclysm.rb` to be opened with Neovim.

- Step-by-step reproduction instructions (by running `brew` commands)
```
brew install fish
exec fish
set EDITOR "/usr/local/bin/nvim"
brew edit cataclysm
```

I've reproduced this with other formulae; and formulae within taps. My fish config is [here](https://github.com/LEARAX/dotfiles/tree/macos/.config/fish). I'm guessing this has something to do with Fish's environment variables; please let me know what other information is needed.

**Comment by jakeydoodle123** [comment]
TLDR;
use:
`export HOMEBREW_EDITOR="/usr/bin/nano"`

When digging around I found that using `set EDITOR "/usr/bin/nano"` would set the environment in fish but it wouldn't read it in brew, then I tried `export EDITOR="/usr/bin/nano"` but it still didn't like this. The error it threw said that I could also try changing HOMEBREW_EDITOR, once I set that using export it worked perfectly fine. I went through and found the line where it loads in the editor from and added EDITOR to it so that it would also work and I'm creating a pull request soon. The issue with it not reading it with `set` might have something to do with the way it reads the info from fish vs bash but honestly I'm not sure.
